netplan/cli/commands/set.py

```diff
@@ -73,7 +74,12 @@ def command_set(self):
             tmp.flush()
             tmp.seek(0, io.SEEK_SET)
             parser.load_nullable_fields(tmp)
-            parser.load_yaml_hierarchy(self.root_dir)
+            if self.origin_hint:  # update only this specific file
+                path = os.path.join(self.root_dir, 'etc', 'netplan', filename)
+                if os.path.isfile(path):
+                    parser.load_yaml(path)
+            else:
+                parser.load_yaml_hierarchy(self.root_dir)
```

Comment:
issue: this won't work in the general case

The only reason this works now is because you have standalone definitions. If your new definition refers to an interface defined in another file, `netplan set` will fail.

A quickfix would be to add a field to netdefs (in the `private_data` field) to mark them as "touched" when parsing a file without name (`npp->current.file_path == NULL`), and dump them out in `write_yaml_file` (or add another variant of this function with these semantics).

The problem with this quickfix is that you'll get the entire netdef dumped, not just the patched parts. A more involved fix would be to change the dirtiness tracker to store origin file rather than a simple boolean. This would allow for fine-grained control over which field we write out, but the implementation could get tricky (or *very* memory-hungry)

Reply:
Thank you very much for your review and hints. I need to allocate some extra time to dive deeper into this and fix it properly!